tests/Common/TokensTest.php

```diff
@@ -1045,4 +1045,114 @@ public function testTokenWithoutTokensManagePermissionCanListOnlyOwnTokenEvents(
             $this->assertEquals(403, $e->getCode());
         }
     }
+
+    /**
+     * @dataProvider limitedTokenOptionsData
+     */
+    public function testGuestTokenCreateLimitedToken(TokenCreateOptions $options)
+    {
+        $client = $this->getGuestClient();
+
+        $creatorToken = $client->verifyToken();
+
+        $this->assertTrue($creatorToken['isMasterToken']);
+        $this->assertFalse($creatorToken['canManageTokens']);
+
+        $tokenId = $client->createToken($options);
+
+        $token = $this->_client->getToken($tokenId);
+
+        if ($options->getDescription()) {
+            $this->assertSame($options->getDescription(), $token['description']);
+        } else {
+            $this->assertSame(sprintf('Created by %s', $creatorToken['description']), $token['description']);
+        }
+
+        $this->assertFalse($token['isMasterToken']);
+        $this->assertFalse($token['canManageBuckets']);
+        $this->assertFalse($token['canManageTokens']);
+        $this->assertFalse($token['canReadAllFileUploads']);
+        $this->assertFalse($token['canPurgeTrash']);
+        $this->assertArrayNotHasKey('componentAccess', $token);
+        $this->assertNotEmpty($token['expires']);
+        $this->assertSame([], $token['bucketPermissions']);
+    }
+
+    public function limitedTokenOptionsData()
+    {
+        return [
+            [
+                (new TokenCreateOptions())
```

Comment:
Když už bys do toho sahal, tak můžeš tady přidat klíče tomu poli, aby bylo jasné co ta věc testuje (tj. "wihtout description" nebo "not allowed optiosn are ignored")